src/main/webapp/search.js

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const URL_STRUCTURE = 'https://www.googleapis.com/youtube/v3/search?part=snippet&type=video&maxResults=5&q=';
+const YOUTUBE_API_KEY = 'AIzaSyDYfjWcy1hEe0V7AyaYzgIQm_rT-9XbiGs';
+
+/**
+ * Retrieve the video result from youtube url.
+ */
+function getVideoResults() {
+  const tabResult = document.getElementById('tab_list');
+  const contentResult = document.getElementById('info_list');
+
+  // Add html for tab bar and content bar
+  tabResult.innerHTML = '';
+  contentResult.innerHTML = '';
+
+  url = getRequestUrl();
+  fetch(url).then((response) => response.json()).then((data) => {
+    videos = data.items;
+    videos.forEach(function(video) {
+      tabResult.appendChild(addVideoTitle(video));
```

Comment:
Generally it can be expensive (client-side resource wise) to append elements to the DOM iteratively. The reason is that whenever the DOM is updated, the browser needs to redrawn the page to display the new elements. 

So in this case, the entire page will need to be redrawn after each `videoTitle` or `videoInfo` is created.

This may result in some weird UX (i.e. a partial result state is displayed to users) +  may slow down the browser.

Instead, what you can do is make use of `documentFragments` so that the page is redrawn only after the HTML for all list elements is calculated.

You can refer to the official [documentation](https://developer.mozilla.org/en-US/docs/Web/API/Document/createDocumentFragment)